jvb/src/test/kotlin/org/jitsi/videobridge/AudioSubscriptionManagerTest.kt

```diff
@@ -0,0 +1,461 @@
+/*
+ * Copyright @ 2025 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.videobridge
+
+import io.kotest.core.spec.IsolationMode
+import io.kotest.core.spec.style.ShouldSpec
+import io.kotest.matchers.collections.shouldBeEmpty
+import io.kotest.matchers.collections.shouldContainExactlyInAnyOrder
+import io.kotest.matchers.shouldBe
+import io.mockk.every
+import io.mockk.mockk
+import org.jitsi.videobridge.message.ReceiverAudioSubscriptionMessage
+import org.jitsi.videobridge.relay.AudioSourceDesc
+
+class AudioSubscriptionManagerTest : ShouldSpec() {
+    override fun isolationMode(): IsolationMode? = IsolationMode.InstancePerLeaf
+
+    val conference: Conference = mockk {
+        every { audioSourceDescs } returns listOf(
+            AudioSourceDesc(1001L, "endpoint1", "source1"),
+            AudioSourceDesc(1002L, "endpoint2", "source2"),
+            AudioSourceDesc(1003L, "endpoint3", "source3")
+        )
+    }
+
+    private val manager = AudioSubscriptionManager(conference)
+
+    init {
+        context("Basic subscription management") {
+            should("handle single endpoint subscription") {
+                val subscription = ReceiverAudioSubscriptionMessage.Include(listOf("source1", "source2"))
+                manager.setEndpointAudioSubscription("endpoint1", subscription)
+
+                // Endpoint should want included sources
+                manager.isEndpointAudioWanted("endpoint1", 1001L) shouldBe true // source1
```

Comment:
Any implications from the fact that an endpoint can subscribe to its own sources? Packets won't be routed back to it, but maybe something more subtle? I'm just wondering

Reply:
If it's with Include type, there will be unnecessary element in the subscribedLocalSources. That's pretty much the only effect I could come up with. No big effect I assume